src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -105,8 +105,16 @@ public function match(\$pathinfo)
     {
         \$allow = array();
         \$pathinfo = rawurldecode(\$pathinfo);
+        \$trimmedPathinfo = rtrim(\$pathinfo, '/');
         \$context = \$this->context;
         \$request = \$this->request;
+        \$requestMethod = \$isLikeGetMethod = \$context->getMethod();
+        \$schema = \$context->getScheme();
+
+        if ('HEAD' === \$requestMethod) {
+            \$isLikeGetMethod = 'GET';
```

Comment:
Actually, we don't want a boolean, because then we can't perform a meaningful optimisation. We just want to have a interpreted method, which we can check against. Otherwise we'd need to do more check than less.